lib/python_helper_sympy.py

```diff
@@ -28,19 +31,60 @@ def __init__(self):
             'cos': sympy.cos,
             'sin': sympy.sin,
             'tan': sympy.tan,
+            'sec': sympy.sec,
+            'csc': sympy.csc,
+            'cot': sympy.cot,
             'arccos': sympy.acos,
             'arcsin': sympy.asin,
             'arctan': sympy.atan,
-            'acos': sympy.acos,
-            'asin': sympy.asin,
-            'atan': sympy.atan,
-            'arctan2': sympy.atan2,
             'atan2': sympy.atan2,
             'exp': sympy.exp,
             'log': sympy.log,
+            'ln': lambda x: sympy.log(x, sympy.E),
             'sqrt': sympy.sqrt,
+            're': sympy.re,
+            'im': sympy.im,
+            'max': sympy.Max,
+            'min': sympy.Min,
         }
 
+    def get_supported_function_names(self):
+        return list(self.functions.keys())
+
+    def add_new_operator_names(self, opNamesList):
+        for opName in list(opNamesList):
+            if opName not in self.functions.keys():
+                sympyDummyFunc = sympy.Function(opName) 
+                self.functions.update({ opName : sympyDummyFunc })
```

Comment:
```suggestion
        for opName in opNamesList:
            if opName not in self.functions:
                self.functions[opName] = sympy.Function(opName)
```
This is a bit of a nitpick, to make the code simpler.